R/jobUtilities.R

```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
   metadata <- readMetadataBlob(jobId)
 
   if (!is.null(metadata)) {
+    job <- getJob(jobId, verbose = FALSE)
+
+    if (job$jobState == "active") {
+      stop(sprintf("job %s has not finished yet, please try again later",
+                   job$jobId))
+    } else if (job$jobState != "completed") {
+      stop(sprintf(
+        "job %s is in %s state, no job result is available",
+        job$jobId,
+        job$jobState
+      ))
+    }
+
+    # if the job has failed task
+    if (job$tasks$failed > 0) {
+      if (metadata$errorHandling == "stop") {
+        stop(
+          sprintf(
+            "job %s has failed tasks and error handling is set to 'stop', no result will be avaialble",
+            job$jobId
+          )
+        )
+      } else {
+        if (job$tasks$succeeded == 0) {
+          stop(sprintf(
+            "all tasks failed for job %s, no result will be avaialble",
+            job$jobId
+          ))
+        }
+      }
+    }
+
     if (metadata$enableCloudCombine == "FALSE") {
-      cat("enalbeCloudCombine is set to FALSE, no job merge result is available",
+      cat("enableCloudCombine is set to FALSE, we will merge job result locally",
           fill = TRUE)
 
-      return()
-    }
+      results <- vector("list", job$tasks$completed)
```

Comment:
This doesn't work if the job is still running?

Reply:
we would error out earlier if the job is still running and will not reach here.